controllers/front/syncAll.php

```diff
@@ -56,6 +56,22 @@ public function __construct()
         $this->syncProductCron = new ShoppingfeedSyncProductModuleFrontController();
     }
 
+    public function checkAccess()
+    {
+        header('Content-type: text/plain');
+
+        if (Tools::getValue('secure_key') == $this->module->secure_key) {
+            return true;
+        }
+
+        if (false == empty((new ShoppingfeedToken())->findByToken(Tools::getValue('token', '')))) {
+            return true;
+        }
+
+        $return = ['success' => false, 'error' => 'Authentication failed'];
+        $this->ajaxDie(json_encode($return));
```

Comment:
ajaxDie not available on 1.6.0. has SF module a Backward compatibility on that ?

Reply:
There is no compatibility.
Pay attention, this method is used on the level of classlib
https://github.com/shoppingflux/module-prestashop/blob/develop/vendor/totpsclasslib/src/Extensions/ProcessMonitor/Controllers/Front/CronController.php#L80

So, I find it is better to provide compatibility on the lever of classlib.
At least on the branch, we use 'feature/34626_diagnostic'

Reply:
You are right. There is an issue on this piece of classlib code.
'feature/34626_diagnostic' is a none stable version.

Could you fix your part of code. I'll talk to classlib team to fix to in the right branch.

Reply:
I have already fixed this problem one year ago
https://github.com/shoppingflux/module-prestashop/blob/develop/classes/ShoppingfeedCronController.php#L33
Sorry for confusing you